### PR TITLE
Editor: Allow parent page to be removed.

### LINKED ISSUE #741: Editor: Can't remove page parent
When editing a page, once you define a parent for it in the Page Options, there is no way to uncheck all radio buttons and bring the page back at the top level.

### PR DESCRIPTION
**Pull request by timmyc**
Fixes #741

Currently in the editor there is no way to remove a page parent if one is currently set.  This branch adds a checkbox above the Parent Page Selector, much like we show in the Add Category screen, to allow removing the parent page option, or setting it back to a "Top level page.".

edit_page_ _testingtimmy2_wordpress_com_ _wordpress_com

__To Test__
- Open up a new page in the editor
- Expand the Page Options accordion, note that "Top level page" is checked
- Add a title and some content
- Select a page parent, note the checkbox is un-checked
- Save the page, and do a hard refresh, check that your parent page selection is persisted
- Check the Top Level Page checkbox, note that the parent page radio is de-selected
- Save the page, hard refresh and note that the page is still set to Top Level

### CODE REVIEW

File: client/post-editor/editor-page-parent/index.jsx [hunk 1]
```diff
@@ -22,25 +24,30 @@ export default React.createClass( {
...
 	render() {
 		return (
 			<AccordionSection className="editor-page-parent">
-				<label>
+				<FormLabel>
 					<span className="editor-page-parent__label-text">{ this.translate( 'Parent Page' ) }</span>
-				</label>
+				</FormLabel>
+				<FormLabel>
+					<FormCheckbox ref="topLevel" checked={ ! this.props.parent } onChange={ this.updatePageParent } />
```
Comment (mtias) [inline client/post-editor/editor-page-parent/index.jsx]: We should use a toggle for these singular controls.

Comment (timmyc) [inline client/post-editor/editor-page-parent/index.jsx]: :+1:

**Comment by folletto** [comment]
Would be possible to add that as an option inside the box, at the top, with a house icon? The reason is that in terms of structure you're just positioning it to the very top, so it's a "top" option in the parent page structure (root).

screen shot 2015-11-27 at 15 56 32

Not sure if it's better to indent or not there. Opinions?

**Comment by timmyc** [comment]
In 002239f use of the toggle, and styling that is currently seen in "Post Status" was added:

top-level

@folletto I believe we might have originally gone that route with the Add Category screen too, and I seem to think we did not go that way due to the nature of checkboxes/multi-select.  Here though, that approach could work with the radios.

I kind of like this toggle approach myself.

**Comment by folletto** [comment]
The item inside the list has two big advantages:

1. It's semantically correct, being the root element
2. It's visually cleaner

Unless there's a big technical issue, we should really try to get it inside the list. :)

**Comment by timmyc** [comment]
@mtias do you have an opinion ^ I can create a prop for an empty/null option in `<PostSelector />` to accomplish this.  @folletto having that as an option inside the selector, would it be hidden or shown when search results are displayed?

**Comment by mtias** [comment]
@timmyc I'd get the top level toggle in so we have something and then look at including it within the list, since it won't be as trivial.

**Comment by timmyc** [comment]
Sounds like a good plan @mtias

**Comment by folletto** [comment]
Agreed :+1:

**Comment by jkudish** [comment]
Quick notes from a quick test sesh:
* it's kinda weird that it's a toggle but i can't untoggle it
* I selected a page parent, then changed my mind and set it back to top level and clicked published, but it published with that initial selection of a parent page
* After I published the page, I had the option of setting the page parent to itself, which is weird. When I selected it, it reverted to top level and that option went away.
* Not sure if related or not, but after making a few rounds of updates I got this (no console errors):

screenshot from 2015-12-08 08-50-02

**Comment by timmyc** [comment]
@jkudish I can't seem to re-create the issue above where you select a page, then click the top level toggle and the old selection is persisted.  Additionally I'm not seeing the newly created page being shown as an option in the selector.

I suppose there might be a brief moment when the post-edit-store does not yet have an ID for the post - and possibly a new `post-list-store` payload comes down with the new page inside of it - but I really can't think of how that might happen.

As for the draft save failure - I added an extra check to `updatePageParent()` in e0f5536 to make sure the argument being passed in is a page with an ID.  The method was taking the event argument and passing `undefined` to the API which was making things :boom:

**Comment by jkudish** [comment]
I just re-tested. Looking better. I can no longer repro 2/3 of the bugs from above.

I can still repro this one:

> After I published the page, I had the option of setting the page parent to itself, which is weird. When I selected it, it reverted to top level and that option went away.

This time I got it by editing a published page which had a parent page and reverting it to a draft.

---

One other bug I just noticed, though maybe it's for a separate PR: the previewed permalink of a page with a parent should be updated to reflect its page parent. If I have a `Page B` with `Page A` as its parent, the permalink is `http://example.wordpress.com/page-a/page-b` but currently the editor shows the permalink as `http://example.wordpress.com/page-b`.

**Comment by timmyc** [comment]
The slug thing is a definite regression - that was working at one point during the great slug explorations of 2015.  I will spawn an issue for that.

As for the page being shown as an option in the parent list - I finally was able to reproduce this.  I had to let the editor site idle for a few minutes for this to happen though.  But it appears another component polls the post-list-store:

`https://public-api.wordpress.com/rest/v1.1/sites/91836948/posts?http_envelope=1&status=publish,private&type=page&site_visibility=visible&modified_after=2015-12-09T20:31:33+00:00`

The key with this request is it does not include `exclude_tree` with a value of the `page.ID`.  The api then returns the page in this request since it matches the query and is not excluded.  This particular bug is kind of out of scope of this PR - but I am guessing it is either the scheduler which seems to be enabled now on pages .

Though, this bug would be fixed by #708 since each component would have their own `post-list-store` to work against then.

**Comment by timmyc** [comment]
@jkudish mind taking one last look at this - well no additional commits but I created a new issue to track the parent slug bit - and explained the other issue you were seeing above.

File: client/post-editor/editor-page-parent/index.jsx [hunk 2]
```diff
@@ -22,25 +24,36 @@ export default React.createClass( {
...
+
+		return this.translate( 'You have no pages yet.' );
 	},
 
 	render() {
 		return (
 			<AccordionSection className="editor-page-parent">
-				<label>
+				<FormLabel>
 					<span className="editor-page-parent__label-text">{ this.translate( 'Parent Page' ) }</span>
```
Comment (aduth) [inline client/post-editor/editor-page-parent/index.jsx]: Unrelated to this pull request, but seems like we could move the `className` to the `<FormLabel>` component and have it work just as well.

File: client/post-editor/editor-page-parent/style.scss [hunk 3]
```diff
@@ -46,3 +46,20 @@
...
+	font-size: 13px;
+	line-height: 1.7;
+}
+
+.editor-page-parent__top-level {
+	align-content: center;
+	display: flex;
+	justify-content: space-between;
+
+	.gridicon {
```
Comment (aduth) [inline client/post-editor/editor-page-parent/style.scss]: I can't find a `<Gridicon />` that this would apply to?

**Comment by aduth** [comment]
>Though, this bug would be fixed by #708 since each component would have their own `post-list-store` to work against then.

Hmm, I'm still seeing this bug even after having rebased against latest master, with #708 having been merged.

**Comment by timmyc** [comment]
> Hmm, I'm still seeing this bug even after having rebased against latest master, with #708 having been merged.

That is because #708 kept all usage of `post-list-store` in the `default` sense ( so shared ).  I will add a change to this PR to scope it to the `post-selector`.

File: client/post-editor/editor-page-parent/style.scss (hunk 3 above)
Comment (timmyc) [inline client/post-editor/editor-page-parent/style.scss]: Good catch, that is a remnant of a prior iteration in this branch :)

**Comment by timmyc** [comment]
@aduth utilizing the new `post-list-store` factory abilities in 29213e6 and removed unused className in 7ed3a2b

**Comment by jkudish** [comment]
On a site with only a few pages, I don't the box around the pages, I am guessing this is expected:

screenshot from 2015-12-11 17-29-53

Other than that (which I don't think is a bug), and the noted issues above, I think this is good to go.

**Comment by mtias** [comment]
> On a site with only a few pages, I don't the box around the pages, I am guessing this is expected:

Yes, few pages don't require the box nor search. Same with few categories.

**Pull request merged**